app/helpers/submissions_helper.rb

```diff
@@ -254,7 +254,8 @@ def remark_complete_but_unreleased(submission)
 
   # Checks if all the assignments for the current submission are marked.
   def all_assignments_marked?
-    Assignment.find(params[:assignment_id]).groupings.all? do |grouping|
+    Assignment.includes(:groupings => :current_submission_used).find(
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Pass `&:marking_completed?` as an argument to `all?` instead of a block.